context/app/static/js/pages/Dataset/hooks.ts

```diff
@@ -171,7 +172,7 @@ function useProcessedDatasetsSections(): { sections: TableOfContentsItem | false
       ? {
           ...getSectionFromString('processed-data'),
           items: searchHits.map((hit) =>
```

Comment:
I don't think it's immediately necessary as we're not facing any performance issues, but we could just generate the section titles while iterating over `searchHits` here to avoid the O(n^2) complexity by also iterating over `searchHits` inside `getProcessedDatasetSection`.